src/isa/exec.rs

```diff
@@ -470,7 +470,9 @@ impl InstructionSet for ArithmeticOp {
                 idx,
                 regs.get(reg, idx).and_then(|val| {
                     let mut n = Number::from(*step);
-                    n.reshape(val.layout());
+                    if !n.reshape(val.layout()) {
+                        unreachable!("reshape target byte length is always greater")
+                    }
```

Comment:
good check!

BTW you can use more simple form
`debug_assert!(n.reshape(val.layout()), "reshape target byte length is always greater");`

Using `debug_assert` allows to get more efficient code in releases (they are cut by compiler for release profiles, avoiding unnecessary jumps in the code)

Reply:
I was not aware of it. thanks